cogs/commands/chatgpt.py

```diff
@@ -176,6 +184,79 @@ async def fetch_previous(
             return await message.channel.fetch_message(message.reference.message_id)
         return None
 
+    @tasks.loop(minutes=30)
+    async def update_channel_descriptions(self):
+        """
+        Update the AI chat description to include the current API usage.
+        Runs in a task loop, firing periodically
+        """
+        channels = (
+            discord.utils.get(self.bot.get_all_channels(), id=c)
+            for c in CONFIG.AI_CHAT_CHANNELS
+        )
+        spent_usd = await self.get_api_usage() / 100
+
+        # we need to do currency conversion
+        async with aiohttp.ClientSession() as session:
+            resp = await session.get(
+                "https://api.exchangerate.host/convert?from=USD&to=GBP"
+            )
+            rate = (await resp.json())["result"]
+
+        spent_gbp = round(spent_usd * rate, 4)
+
+        for ch in channels:
+            # fetches all channels, we only want to edit them if they're text channels
```

Comment:
Consider only editing (e.g.) only the first channel in the list -- we allow Apollo to chat in #bot-commands too, but we don't need it edited too. Could add another config parameter but that's unwieldy imo